compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/gestures/Scrollable.kt

```diff
@@ -953,6 +953,9 @@ internal class DefaultFlingBehavior(
                         lastAnimationCycleCount++
                     }
                 } catch (exception: CancellationException) {
+                    // This is where the ScrollCancellationException should be caught,
+                    // but the parent class CancellationException is caught here for
+                    // backwards compatibility.
```

Comment:
What is "backwards compatibility" here? What cases it covers? Aren't all references in the same module and already updated?

Reply:
I mean, we keep handling `CancellationException` here to catch new potential cases. For example if someone will make similar trick at another place.

Reply:
I don't think that we should keep it for that. I hope this will be upstreamed soon to avoid such changes during the merge.
However, I'm not going to block the PR because of this.

cc @igordmn

Reply:
If you want to create upstream PR in the close future, let's change. Otherwise it's safer to keep as it is.